pkg/pub_dartdoc/test/kind_names_test.dart

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:dartdoc/dartdoc.dart';
+import 'package:pub_dartdoc_data/dartdoc_index.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test('kind names', () {
+    final expected = <int, String>{};
+    for (final k in Kind.values) {
+      expected[k.index] = k.toString();
+    }
+    expect(kindNames, expected);
```

Comment:
Is this supposed to test that `kindNames` as defined in `pkg/pub_dartdoc_data/lib/dartdoc_index.dart` matches `Kind.values` ?

Can we add a comment.

-------------------

The reason you don't want to use `Kind.values` directly, is that you are not sure that `k.toString()` will work in production?

Reply:
The reason was that this parsing code should not depend on `package:dartdoc` in the long term (if we want to call SDK-specific dartdoc and/or non-customized dartdoc), but we still need these values, and matching it here as a separate test seemed like a good in-between.